docs-client/src/containers/MethodPage/index.tsx

```diff
@@ -136,7 +136,36 @@ type Props = OwnProps &
   }>;
 
 const MethodPage: React.FunctionComponent<Props> = (props) => {
-  const [debugFormIsOpen, setDebugFormIsOpen] = React.useState(false);
+  const [debugFormIsOpen, setDebugFormIsOpenState] = React.useState(false);
+
+  const { location, history } = props;
+  const setDebugFormIsOpen: React.Dispatch<SetStateAction<boolean>> =
+    React.useCallback(
+      (value) => {
+        const valueToSet =
+          value instanceof Function ? value(debugFormIsOpen) : value;
+        const urlParams = new URLSearchParams(location.search);
+        if (valueToSet === true) {
+          urlParams.set('debug_form_is_open', `${valueToSet}`);
+        } else {
+          urlParams.delete('debug_form_is_open');
+        }
+
+        const serializedParams = `?${urlParams.toString()}`;
+        if (serializedParams !== location.search) {
+          history.push(`${location.pathname}${serializedParams}`);
+        }
+
+        return setDebugFormIsOpenState(valueToSet);
+      },
+      [
+        debugFormIsOpen,
+        setDebugFormIsOpenState,
+        history,
+        location.search,
+        location.pathname,
+      ],
+    );
```

Comment:
We are manually setting all headers in Debug page submit so copied to logic from there. But I believe we can use some param hooks provided by router. Not 100% sure, don't have much time so wanted to fix the bug ASAP 😓